File: main/angry-professor/angry-professor.py
#!/usr/bin/env python3

def read_record():
    return map(int, input().split())

def is_class_canceled(arrivals, needed):
    return sum(1 for time in arrivals if time <= 0) < needed

for _ in range(int(input())):
    _, needed = read_record()
    arrivals = read_record()
    print('YES' if is_class_canceled(arrivals, needed) else 'NO')

File: main/angry-professor/angry-professor.pl
#!/usr/bin/env perl

use strict;
use warnings;
use Carp qw(croak);

sub read_record {
    return split /\s+/msx, <>;
}

sub is_class_canceled {
    my ($needed, @arrivals) = @_;
    if ($needed <= 0) { return 0 }

    for my $time (@arrivals) {
        if ($time <= 0 && --$needed == 0) { return 0 }
    }

    return 1;
}

for (1..<>) {
    my ($len, $needed) = read_record;
    my @arrivals = read_record;
    scalar @arrivals == $len or croak 'wrong record length';

    print is_class_canceled($needed, @arrivals) ? "YES\n" : "NO\n";
}
